### PR TITLE
Colorado Sales Tax Refund

### LINKED ISSUE #2915: Add Colorado state sales tax refund
Line 33 on the Colorado income tax form has a credit for Colorado state sales tax that is missing in the 0.446.0 version of PolicyEngine-US.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2915

### Summary
🏷️💰🛍️

This pull request adds the parameters and variables for the Colorado sales tax refund credit, which is a state income tax credit based on the filer's modified adjusted gross income and filing status. The parameters are defined in separate files for each filing status, and the variable for the modified adjusted gross income is defined in `co_modified_agi.py`. The change also includes the description, metadata, and references for the parameters and variables. The purpose of this change is to implement the Colorado sales tax refund credit in the policy engine model.

> _`Sales_tax_refund`_
> _Brackets for each filing type_
> _Autumn leaves fall fast_

### Walkthrough
*  Add parameters for the Colorado sales tax refund amount for different filing statuses and income brackets ([link](https://github.com/PolicyEngine/policyengine-us/pull/2920/files?diff=unified&w=0#diff-af4d1e294b63ede7e25d0b31cead606b98f7e007c6e1e7032e86663dbec6aceaR1-R49), [link](https://github.com/PolicyEngine/policyengine-us/pull/2920/files?diff=unified&w=0#diff-69acb884e1481408d1ad86a3a723fcabfe0e9aea8f1cacbae07471f031e42a6eR1-R49), [link](https://github.com/PolicyEngine/policyengine-us/pull/2920/files?diff=unified&w=0#diff-6a2cfcf7e8a4534205a7b160d2d248e36eb9389058c4fd75dfd7a97b34cb3587R1-R50), [link](https://github.com/PolicyEngine/policyengine-us/pull/2920/files?diff=unified&w=0#diff-e1800f89cc76fc1484bb4e666ba90dded706b948e4233409b03e550a6c942724R1-R49), [link](https://github.com/PolicyEngine/policyengine-us/pull/2920/files?diff=unified&w=0#diff-a3887f95db8346b7c5b52125c555d3a43230c34aedd9bddd8bf753625828d44eR1-R49))
* Add variable for the Colorado modified adjusted gross income for TABOR ([link](https://github.com/PolicyEngine/policyengine-us/pull/2920/files?diff=unified&w=0#diff-6e5db3dd6b3f3ab0818838c35a8b590c1ab625f58def4d6397b3a6f4d5f97b76L1-R15))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/age_threshold.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+description: Colorado limits its sales tax refund to filers at or above this age threshold.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/age_threshold.yaml:1]:
```suggestion
description: Colorado limits its sales tax refund to filers this age or older.
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/head_of_household.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+description: Colorados allows for this sales tax refund for head of household filers, based on the modified adjusted gross income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/head_of_household.yaml:1]:
```suggestion
description: Colorado provides this sales tax refund for head of household filers, based on modified adjusted gross income.
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/head_of_household.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/head_of_household.yaml:1]: We can simplify this by following the legal code, which states that the thresholds are the same for joint and others, but joint filers get double the amount. So just make one bracket and then a multiplier varying by filing status.

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/agi_sources.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+description: Colorado accounts for these income sources for the modified adjusted gross income.
+values:
+  2021-01-01:
+    - adjusted_gross_income
+    - tax_exempt_social_security
+    - tax_exempt_interest_income
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/agi_sources.yaml:6]: This assumes that all tax-exempt interest income is from state and local bonds, which I don't think we want to do per the other conversation.

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/agi_sources.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+description: Colorado accounts for these income sources for the modified adjusted gross income.
+values:
+  2021-01-01:
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/agi_sources.yaml:3]: per the law
```suggestion
  1999-01-01:
```

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        head_eligible = age_head >= p.age_threshold
+        age_spouse = tax_unit("age_spouse", period)
+        spouse_eligible = age_spouse >= p.age_threshold
+        income_tax = tax_unit(
+            "co_income_tax_before_non_refundable_credits", period
+        )
+        filing_status = tax_unit("filing_status", period)
+        joint = filing_status == filing_status.possible_values.JOINT
+        age_eligible = where(
+            joint, head_eligible & spouse_eligible, head_eligible
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py:26]: Source on both spouses needing to be 18+? I didn't see that in the form or law.

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.co.tax.income.credits.sales_tax_refund
+        age_head = tax_unit("age_head", period)
+        head_eligible = age_head >= p.age_threshold
+        age_spouse = tax_unit("age_spouse", period)
+        spouse_eligible = age_spouse >= p.age_threshold
+        income_tax = tax_unit(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py:20]: I'd suggest also checking if employment_income > 0, as a proxy for getting a refund for withheld taxes, the other pathway to eligibility. And file an issue to capture this more accurately.
```suggestion
        # Legal code is ambiguous, but the form points to the line corresponding to
        # tax before non-refundable credits.
        income_tax = tax_unit(
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py:26]: I read this as both spouses have to meet the eligibility 

image

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            tax_unit("co_income_tax_before_non_refundable_credits", period) > 0
+        )
+        employment_income_eligible = (
+            add(tax_unit, period, ["employment_income"]) > 0
+        )
+        income_eligible = employment_income_eligible | income_tax_eligible
+        filing_status = tax_unit("filing_status", period)
+        joint = filing_status == filing_status.possible_values.JOINT
+        age_eligible = where(
+            joint, head_eligible & spouse_eligible, head_eligible
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py:32]: @PavelMakarchuk change age eligibility to or

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py:32]: file issue - consolidate

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/multiplier.yaml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+description: Colorado multiplies the base sales tax refund amount by this amount, based on the modified adjusted gross income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/multiplier.yaml:1]:
```suggestion
description: Colorado multiplies the base sales tax refund amount by this amount, based on filing status.
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/multiplier.yaml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+metadata:
+  period: year
+  unit: /1
+  label: Colorado sales tax refund multiplier
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/multiplier.yaml:5]:
```suggestion
  label: Colorado sales tax refund filing status multiplier
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/agi_sources.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/agi_sources.yaml:1]: rename `magi_sources`

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    definition_period = YEAR
+    reference = "https://tax.colorado.gov/sites/tax/files/documents/DR_0104_Book_2022.pdf#page=23"
+    defined_for = StateCode.CO
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.co.tax.income.credits.sales_tax_refund
+        age_head = tax_unit("age_head", period)
+        head_eligible = age_head >= p.age_threshold
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py:17]: and spouse
```suggestion
        head_age_eligible = age_head >= p.age_threshold
```

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        age_spouse = tax_unit("age_spouse", period)
+        spouse_eligible = age_spouse >= p.age_threshold
+        # Legal code is ambiguous, but the form points to the line corresponding to
+        # tax before non-refundable credits.
+        income_tax_eligible = (
+            tax_unit("co_income_tax_before_non_refundable_credits", period) > 0
+        )
+        employment_income_eligible = (
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund_eligible.py:25]:
```suggestion
        # Filers that had Colorado tax withheld are also eligible.
        # Use employment income as a proxy.
        employment_income_eligible = (
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**